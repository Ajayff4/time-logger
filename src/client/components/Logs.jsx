import React, { Component } from 'react'

class Logs extends Component {
    tableData = [
        {
            "key": "#",
            "tag": "Tag",
            "category": "Category",
            "time": "Time",
            "duration": "Duration",
            "details": "Details",
            "operations": "Operations"
        },
        {
            "key": 1,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "5min",
            "details": "--"
        },
        {
            "key": 2,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "10min",
            "details": "--"
        },
        {
            "key": 3,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "15min",
            "details": "--"
        },
        {
            "key": 4,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "20min",
            "details": "--"
        },
        {
            "key": 5,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "25min",
            "details": "--"
        },
        {
            "key": 6,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "30min",
            "details": "--"
        },
        {
            "key": 7,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "35min",
            "details": "--"
        },
        {
            "key": 8,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "40min",
            "details": "--"
        },
        {
            "key": 9,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "45min",
            "details": "--"
        },
        {
            "key": 10,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "50min",
            "details": "--"
        },
        {
            "key": 11,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "55min",
            "details": "--"
        },
        {
            "key": 12,
            "tag": "sleeping",
            "category": "official",
            "time": "8:30 AM",
            "duration": "60min",
            "details": "--"
        }
    ];
    render() {
        let BUTTON_FLAG = false;
        return (
            <>
                <h2>Logs Component</h2>
                <center>
                    <div className="container">
                        <table className="table table-dark table-striped log-table-css">
                            <tbody>
                                {

                                    this.tableData.map(row => {
                                        let rowData = [];
                                        rowData.push(<tr>

                                        </tr>)
                                        for (let attribute in row) {
                                            rowData.push(<><td key={attribute + row.id}>{row[attribute]}</td></>);
                                        }
                                        if (BUTTON_FLAG === true)
                                            rowData.push(<td><button id="editButton">Edit</button><button id="deleteButton">Delete</button></td>);
                                        BUTTON_FLAG = true;
                                        return (<tr key={row.id}>{rowData}</tr>);
                                    })
                                }
                            </tbody>
                        </table>
                    </div>
                </center>
            </>
        )
    }
}

export default Logs;