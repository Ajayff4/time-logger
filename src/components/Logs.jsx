import React, { Component } from 'react'
import axios from 'axios'
import * as actions from '../actions'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

class Logs extends Component {
    static propTypes = {
        getAllLogs: PropTypes.func.isRequired,
        failedToFetchData: PropTypes.func.isRequired
    }    

    constructor(props) {
        super(props)
        this.state = {
            logs: []
        }
    }

    fetchAllLogs = () => {
        axios.get("http://localhost:5000/rest/logs")
        .then(res => {
            this.setState({logs: res.data.filter((log) => { return this.props.username===log.username})})
            this.props.getAllLogs(this.state.logs)
        })
        .catch(err => {
            this.props.failedToFetchData(`${err}`);
        })
    }

    componentDidMount() {
        this.fetchAllLogs();
    }

    handleCompletedLog = (logId) => {
        axios.put(`http://localhost:5000/rest/logs/${logId}`)
        .then(res=> {
            alert("log completed" + logId)
            this.fetchAllLogs();
            this.props.completedLog();
        }).catch(err => {
            this.props.failedToFetchData(`${err}`);
        })
    }

    handleDeleteLog = (logId) => {
        axios.delete(`http://localhost:5000/rest/logs/${logId}`,
        {
            headers: {"Access-Control-Allow-Origin": "*"}
        })
        .then(res => {
            if(res.data != null) {
                alert(`logId ${logId} deleted!!!`)
                this.setState({
                    logs: this.state.logs.filter(log => log.id !== logId)
                })
                this.props.deletedLog()
            }
        }).catch(err => {
            this.props.failedToFetchData(`${err}`);
        })
    }

    render() {
        let tableData = []
        tableData.push(
            <tr key={"header"}>
                <td>Id</td>
                <td>User</td>
                <td>Tag</td>
                <td>Date</td>
                <td>Time</td>
                <td>Duration</td>
                <td>Log Details</td>
                <td>Opeartions</td>
            </tr>
        )
        // eslint-disable-next-line
        this.state.logs.map((log, i) => {
            let row = []
            let logId=undefined;
            let isCompleted = false;
            for (let field in log) {
                if(field==="id"){ logId = log[field] }
                if(field==="completed"){ isCompleted = log[field] }
                if(field!=="completed"){ row.push(<td key={field + log[field]}>{log[field]}</td>) }
            }
            row.push(
                <td key={i} id={"operations"}> 
                    <button id={isCompleted?"completedButtonOk":"completedButton"} disabled={isCompleted} name="completed" onClick={() => {this.handleCompletedLog(logId)}}>
                        {isCompleted?"Completed":"Click Me"}
                    </button>
                    <button id="deleteButton" name="delete" onClick={() => this.handleDeleteLog(logId)}>
                        Delete
                    </button>
                </td>
            )
            tableData.push(<tr key={logId}>{row}</tr >)
        })

        return (
            <>
            <div id="container">
                <h2>Logs Component</h2>
                <center>
                    {this.state.logs!==undefined && this.state.logs.length>0 ?
                    <div id="table-wrapper">
                        <div id="table-scroll">
                            <div className="container">
                                <table className="table table-dark table-striped">
                                    <tbody>
                                        {tableData}
                                    </tbody>
                                </table>
                            </div>
                        </div>
                    </div>
                    :
                    <h3>No Logs were found</h3>
                }
                    
                </center>
            </div>
            </>
        )
    }
}

const mapStateToProps = state => {
    return {
        username: state.user.username,
        logs: state.log.logs
    }
}

const mapDispatchToProps = dispatch => {
    return {
        failedToFetchData: (err) => dispatch(actions.failedToFetchData(err)),
        completedLog: () => dispatch(actions.completedLog()),
        deletedLog: () => dispatch(actions.deletedLog()),
        getAllLogs: (logs) => dispatch(actions.getAllLogs(logs)),
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Logs)