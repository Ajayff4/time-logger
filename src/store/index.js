import { createStore, applyMiddleware } from 'redux'
import rootReducer from '../reducers/index'
//import logger from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension'
import rootSaga from '../saga'
import createSagaMiddleware from 'redux-saga'

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(sagaMiddleware))
)

sagaMiddleware.run(rootSaga)

export default store