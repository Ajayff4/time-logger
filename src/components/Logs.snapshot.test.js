import React from 'react'
import renderer from 'react-test-renderer'
import Logs from './Logs'
import { Provider } from 'react-redux';
import store from '../store';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Logs Snapshot Testing...', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(<Provider store={store}><Router><Logs /></Router></Provider> ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});