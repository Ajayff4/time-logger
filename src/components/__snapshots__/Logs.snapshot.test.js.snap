// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logs Snapshot Testing... renders component correctly 1`] = `
<div
  id="container"
>
  <h2>
    Logs Component
  </h2>
  <center>
    <h3>
      No Logs were found
    </h3>
  </center>
</div>
`;
